include/deal.II/base/big_mpi_compat.h

```diff
@@ -0,0 +1,331 @@
+          return ierr;
+        }
+
+      ierr = MPI_Type_free(&chunks);
+      if (ierr != MPI_SUCCESS)
+        {
+          std::cerr << "MPI_Type_contiguous() call failed!" << std::endl;
+          return ierr;
+        }
+      ierr = MPI_Type_free(&remainder);
+      if (ierr != MPI_SUCCESS)
+        {
+          std::cerr << "MPI_Type_contiguous() call failed!" << std::endl;
+          return ierr;
+        }
+
+#    ifndef MPI_COMPAT_SKIP_SIZE_CHECK
+      MPI_Count size_new;
+      ierr = MPI_Type_size_x(*newtype, &size_new);
+
+      if (size_old * count != size_new)
+        {
+          std::cerr
+            << "MPI_Type_contiguous_c() produced an invalid result. Expected = "
+            << size_old << " * " << count << " = " << size_old * count
+            << " but received " << size_new << std::endl;
+          return MPI_ERR_INTERN;
+        }
+#    endif
+
+      return MPI_SUCCESS;
+    }
+}
+
+/**
+ * Send a package to rank @p dest with a (possibly large) @p count.
+ *
+ * See the MPI 4.x standard for details.
+ */
+inline int
+MPI_Send_c(const void * buf,
+           MPI_Count    count,
+           MPI_Datatype datatype,
+           int          dest,
+           int          tag,
+           MPI_Comm     comm)
+{
+  if (count <= BigMPICompat::mpi_max_count)
+    return MPI_Send(buf, count, datatype, dest, tag, comm);
+
+  MPI_Datatype bigtype;
+  int          ierr;
+  ierr = MPI_Type_contiguous_c(count, datatype, &bigtype);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+  ierr = MPI_Type_commit(&bigtype);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+
+  ierr = MPI_Send(buf, 1, bigtype, dest, tag, comm);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+
+  ierr = MPI_Type_free(&bigtype);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+  return MPI_SUCCESS;
+}
+
+/**
+ * Receive a package from rank @p source with a (possibly large) @p count.
+ *
+ * See the MPI 4.x standard for details.
+ */
+inline int
+MPI_Recv_c(void *       buf,
+           MPI_Count    count,
+           MPI_Datatype datatype,
+           int          source,
+           int          tag,
+           MPI_Comm     comm,
+           MPI_Status * status)
+{
+  if (count <= BigMPICompat::mpi_max_count)
+    return MPI_Recv(buf, count, datatype, source, tag, comm, status);
+
+  MPI_Datatype bigtype;
+  int          ierr;
+  ierr = MPI_Type_contiguous_c(count, datatype, &bigtype);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+
+  ierr = MPI_Type_commit(&bigtype);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+
+  ierr = MPI_Recv(buf, 1, bigtype, source, tag, comm, status);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+
+  ierr = MPI_Type_free(&bigtype);
+  if (ierr != MPI_SUCCESS)
+    return ierr;
+  return MPI_SUCCESS;
+}
+
+#  endif
+
+
+namespace BigMPICompat
+{
+  /**
+   * Write a possibly large @p count of data at the location @p offset.
+   *
+   * See the MPI 4.x standard for details.
+   */
+  inline int
+  MPI_File_write_at_c(MPI_File     fh,
```

Comment:
What is the reason why you are putting these functions into a namespace, but not the ones above?